File: apps/merchant-api/InventoryService/src/InventoryService.Domain/Concretes/Product.cs
using InventoryService.Domain.Bases;

namespace InventoryService.Domain.Concretes;

public class Product : BaseEntity
{
    public Guid? StoreId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public double BasePrice { get; set; }
    public string Brand { get; set; } = string.Empty;
    public int? LowStockThreshold { get; set; }
    public float Rating { get; set; } = 0; 
    public ICollection<Image> Images { get; set; } = new List<Image>();
    public ICollection<ProductVariant> ProductVariants { get; set; } = new List<ProductVariant>();
    public ICollection<Category> Categories { get; set; } = new List<Category>();
    public ICollection<WishList> WishList { get; set; } = new List<WishList>();
}

File: apps/merchant-web/src/commons/entities/concretes/Product.ts
import { UUID } from "crypto";
import ProductVariant from "./ProductVariant";
import Category from "./Category";
import Image from "./Image";
import ProductReview from "./ProductReview";


export default class Product {
  id: UUID;
  storeId: UUID;
  name: string;
  description: string;
  price: number;
  brand: string;
  images: Array<Image>;
  productVariants: Array<ProductVariant>;
  categories: Array<Category>;
  productReviews: Array<ProductReview>;
  lowStockThreshold?: number;

  constructor(
    id: UUID,
    storeId: UUID,
    name: string,
    description: string,
    price: number,
    brand: string,
    images: Array<Image>,
    productVariants: Array<ProductVariant>,
    categories: Array<Category>,
    productReviews: Array<ProductReview>,
    lowStockThreshold?: number,
  ) {
    this.id = id;
    this.storeId = storeId;
    this.name = name;
    this.description = description;
    this.price = price;
    this.brand = brand;
    this.categories = categories;
    this.images = images;
    this.productVariants = productVariants;
    this.productReviews = productReviews;
    this.categories = categories;
    this.lowStockThreshold = lowStockThreshold;
  }
}
